src/interp/interp.cc

```diff
@@ -1144,15 +1127,16 @@ Result Thread::TableGet(const uint8_t** pc) {
 
 Result Thread::TableFill(const uint8_t** pc) {
   Table* table = ReadTable(pc);
-  uint32_t table_size = table->size();
   uint32_t size = Pop<uint32_t>();
   Ref value = static_cast<Ref>(Pop<Ref>());
   uint32_t dst = Pop<uint32_t>();
-  bool ok = ClampToBounds(dst, &size, table_size);
+  bool ok = CheckBounds(dst, size, table->size());
+  if (!ok) {
+    TRAP_MSG(TableAccessOutOfBounds, "table.fill out of bounds");
+  }
```

Comment:
Use `TRAP_UNLESS` and move `CheckBounds` inside it?

Reply:
We could, but this matches the way it's written elsewhere.